File: src/main/java/com/project/food_delivery/dtos/PlaceDataDto.java
package com.project.food_delivery.dtos;

import io.swagger.v3.oas.annotations.Parameter;
import io.swagger.v3.oas.annotations.media.Content;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

@Data
public class PlaceDataDto {
    @Schema(description = "Name", example = "Metro")
    private String name;

    @Parameter(required = true, content = @Content(
            mediaType = "application/json",
            schema = @Schema(implementation = AddressDto.class)))
    private AddressDto address;

    @Schema(description = "Place category", example = "Supermarket")
    private String placeCategory;
}

File: front-end/src/app/common/place-data-dto.ts
import {AddressDto} from "./address-dto";

export class PlaceDataDto {
    public name: string;

    public address: AddressDto;

    public placeCategory: string;

    constructor(name: string, address: AddressDto, placeCategory: string) {
        this.name = name;
        this.address = address;
        this.placeCategory = placeCategory;
    }
}
